bare-metal/container-linux/kubernetes/profiles.tf

```diff
@@ -1,18 +1,39 @@
+locals {
+  flavor  = "${element(split("-", var.os_channel), 0)}"
```

Comment:
One of the nice properties of #211 is that if a user inputs a value like `stable` (instead of `coreos-stable`), the Terraform error strongly hints at the problem showing the AMI data source and indicating that no matching images were found.

I wouldn't do any contortion to make something similar here, but worth considering.